--- a/drivers/mtd/spi-nor/spi-nor.c	2017-04-24 16:24:47.818868807 +0000
+++ b/drivers/mtd/spi-nor/spi-nor.c	2017-04-24 16:26:59.834870560 +0000
@@ -13,6 +13,7 @@
 #include <linux/err.h>
 #include <linux/errno.h>
 #include <linux/module.h>
+#define DEBUG
 #include <linux/device.h>
 #include <linux/mutex.h>
 #include <linux/math64.h>
@@ -489,6 +490,7 @@
 	{ "n25q128a11",  INFO(0x20bb18, 0, 64 * 1024,  256, 0) },
 	{ "n25q128a13",  INFO(0x20ba18, 0, 64 * 1024,  256, 0) },
 	{ "n25q256a",    INFO(0x20ba19, 0, 64 * 1024,  512, SECT_4K) },
+	{ "n25q256b",    INFO(0x20bb19, 0, 64 * 1024,  512, SECT_4K) },
 	{ "n25q512a",    INFO(0x20bb20, 0, 64 * 1024, 1024, SECT_4K) },
 
 	/* PMC */
--- a/drivers/mtd/devices/m25p80.c	2017-05-15 15:31:16.602917677 +0000
+++ b/drivers/mtd/devices/m25p80.c	2017-05-15 15:36:02.266921470 +0000
@@ -281,7 +281,7 @@
 	{"mx25l12855e"},{"mx25l25635e"},{"mx25l25655e"},{"mx66l51235l"},
 	{"mx66l1g55g"},
 	{"n25q064"},	{"n25q128a11"},	{"n25q128a13"},	{"n25q256a"},
-	{"n25q512a"},	{"n25q512ax3"},	{"n25q00"},
+	{"n25q256b"},	{"n25q512a"},	{"n25q512ax3"},	{"n25q00"},
 	{"pm25lv512"},	{"pm25lv010"},	{"pm25lq032"},
 	{"s25sl032p"},	{"s25sl064p"},	{"s25fl256s0"},	{"s25fl256s1"},
 	{"s25fl512s"},	{"s70fl01gs"},	{"s25sl12800"},	{"s25sl12801"},
